src/sections/TransactionInfoSection.tsx

```diff
@@ -200,4 +198,6 @@ const TransactionInfoSection = () => {
   )
 }
 
+const isTxConfirmed = (tx: TransactionLike): tx is Transaction => true
```

Comment:
Shouldn't the type assertion logic be defined in the function?
You always return `true` in that case. Not sure Typescript can actually compare types at runtime?

Cf: https://stackoverflow.com/questions/44078205/how-to-check-the-object-type-on-runtime-in-typescript

Reply:
I implemented the above code after reading the same sources that you posted. The link to the user-defined type guard is a depricated page on the TypeScript documentation. The "updated" way to implement the same logic, is the one above.

Reply:
[Here's ](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)the up-to-date page. 

I'm confused. It looks to me that `isTxConfirmed` will always return true, as you don't return a type predicate, ie. a piece of logic to say when the input must be considered as a `Transaction` (like checking for a certain field presence etc.)

Reply:
Interesting, now I get redirected to a different page than I was before when clicking on the deprecation warning message!

Reply:
I can't find the source anymore. Besides, I was also baffled by this quite a lot. Initially I started by checking for the presence of `blockHash`, but somehow I ended up there. Updating the code to this:

```tsx
const isTxConfirmed = (tx: TransactionLike): tx is Transaction => {
  return (tx as Transaction).blockHash !== undefined
}
```

I've [verified](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCpTiAzoswD2IyA3gLABQyVyAFnFjQFzJZhSgDmA3BdcgEYAbfAgDWACXpMWbTj0rU8AWwis4SgA7MQAVyX9o8gL4VQkWIhQBVEAkIxgUFQBNUADxK9qdBs1bsQbi8qBDpQADEofCVtPQMoeT5QuFBUfFj9QwoTcmcIBEE4KBQwAE8NFHRMHAQ8QgAZYFEUAF40DGxcAiIAH2QbOxAHJwhXN3kKQdZkYCx3AGF7RxdkNoAKMDdmKs7a7sbmgEpmTZmsduquwlWAPk8FYrAdKCINj3oL3bqQQ4A6IREEikyAAhC02joQHkHCBRtkKJNCNMlKUdjVvtsOujuqsSD5pAByNwEgA0AmEYkkvmQBNKpOQylUYHUWmQAEYclMwMgUQMliNXFiriBmHyhstRu5ccR8cwifTkhEojF2WTFSA0swOQjyFN8IIIL9hBw1rMFvyXGsUWjhYdDojsPrDcbTXM3ItxQKraUxcMXDa9oQ7UA) that this works.

Reply:
I'm actually relieved. This makes sense now 🙃